app/views/contexts/hub_context.py

```diff
@@ -107,28 +110,35 @@ def get_section_url(section_id, list_item_id) -> str:
 
         return url_for("questionnaire.get_section", section_id=section_id)
 
-    def _get_row_for_repeating_section(self, section_id, list_item_id):
+    def _get_row_for_repeating_section(self, section_id, list_item_id, list_item_index):
         repeating_title = self._schema.get_repeating_title_for_section(section_id)
 
         title = self._placeholder_renderer.render_placeholder(
             repeating_title, list_item_id
         )
 
-        return self._get_row_for_section(title, section_id, list_item_id)
+        repeating_section_id = section_id + "-" + str(list_item_index)
 
-    def _get_row_for_section(self, section_title, section_id, list_item_id=None):
+        return self._get_row_for_section(
+            title, repeating_section_id, section_id, list_item_id
+        )
+
+    def _get_row_for_section(
+        self, section_title, section_id, section_title_id, list_item_id=None
```

Comment:
What is being passed here as `section_title_id` should be reverted to be passed as `section_id`, as that is what it is. The new parameter being passed should be simply `list_index` as it is elsewhere, and the concatenation with `section_id` should happen in `get_row_context_for_section`.